src/compositemark.ts

```diff
@@ -72,3 +79,175 @@ add(ERRORBAR, (spec: GenericUnitSpec<ERRORBAR, Encoding>): LayerSpec => {
     ]
   };
 });
+
+add(BOX, (spec: GenericUnitSpec<BOX, Encoding>): LayerSpec => {
+  const {mark: _m, encoding: encoding, ...outerSpec} = spec;
+  const {x: _x, y: _y, ...nonPositionEncoding} = encoding;
+
+  let fullEncoding: any = {};
+
+  let isVertical: boolean;
+
+  if (isDiscrete(encoding.x) && isContinuous(encoding.y)) {
```

Comment:
The code is repeated twice for vertical and horizontal.  

(1) This is hard to read and review.  
(2) This might cause problems in the future when we want to extend box plot syntax.  (We will be forced to write every single logic twice, which makes it very prone for error.) 

I'm sure you can write this with half of the code plus a bit more logic checking.